samples/mem_tg/tg_test.h

```diff
@@ -171,22 +181,85 @@ class tg_test : public test_command
         // Read HW details
 
         if (0 == tg_exe_->mem_speed_) {
-	  tg_exe_->mem_speed_ = 300;
-	  std::cout << "Memory channel clock frequency unknown. Assuming "
-		    << tg_exe_->mem_speed_ << " MHz." << std::endl;
-	}
+          tg_exe_->mem_speed_ = 300;
+          std::cout << "Memory channel clock frequency unknown. Assuming "
+              << tg_exe_->mem_speed_ << " MHz." << std::endl;
+        }
         else {
             std::cout << "Memory clock from command line: "
                       << tg_exe_->mem_speed_ << " MHz" << std::endl;
         }
 
-        auto ret = config_input_options();
-        if (ret != 0) {
-            std::cerr << "Failed to configure TG input options" << std::endl;
-            return ret;
+        // Parse mem_ch_ into array of selected channels and number of channels
+	int *channels = NULL;
+        int num_channels = 0;
+        if (0 < (tg_exe_->mem_ch_).size()) {
+          if ((tg_exe_->mem_ch_[0]).find("all") == 0){	
+            uint64_t mem_capability = tg_exe_->read64(MEM_TG_CTRL);
+            channels = new int[sizeof(uint64_t)]; // size should be same as mem_capability 
+            int last = 0;
+            for (uint32_t i = 0; i < sizeof(uint64_t); i++){  // number of itterations should be same as mem_capability 
```

Comment:
If the loop variable "i" was at a higher scope, then you could eliminate the "last" variable.  The resulting code would be clearer.

Reply:
I think I need 2 variables in either case, `i`  is the iterator to keep track of which bit is being checked and `last` is the index of last bit that is set in the 64-bit value at MEM_TG_CTRL. Although I think I can eliminate last since it gets assigned to num_channels anyway; I'll just increment num_channels instead. Thanks!